flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/BoundedTwoInput.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.operators;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * Interface for the two-input operator that can process EndOfInput event.
+ */
+@PublicEvolving
+public interface BoundedTwoInput {
+
+	/**
+	 * It is notified that no more data will arrive on the input identified
+	 * by {@code inputId}.
+	 */
+	void endInput(InputIdentifier inputId) throws Exception;
```

Comment:
Is there any reason why we want to special-case between one-input and two-input, when this interface can actually work for [1..n(32?)]-inputs? Why not unify into just this one interface?

Reply:
I guess @StefanRRichter is referring to `BoundedTwoInput` class name, that it could be renamed to more general one.

Reply:
In the case of one-input, the interface method does not need to be parameterized,  which seems a bit confusing.

Reply:
For one-input operator, this interface method does not need parameters, and it seems confusing if one parameter is taken.

Reply:
Okay.